chain/chaintest/balance_handler_test_helpers.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec/codectest"
+	"github.com/ava-labs/hypersdk/state/tstate"
+)
+
+// TestBalanceHandler tests [b] by doing the following:
+//
+// - Creating a new instance of tstate with permissions from SponsorStateKeys().
+//   - The test account balance is initialized to 0.
+//
+// - Call AddBalance() to increment the test account balance by 1.
+// - Call GetBalance() and require that the test account balance is 1.
+// - Call CanDeduct() to check if the test account balance can be deducted by 1.
+// - Call Deduct() to decrement the test account balance by 1.
+// - Call GetBalance() and require that the test account balance is 0.
+//
+// TestBalanceHandler fails if any of the above method calls returns an error.
```

Comment:
IMO these are implementation details of the function and don't need to be documented

Reply:
Done (as per offline chat, we now refer developers to the BalanceHandler interface to see the invariants `b` must follow)